Fitness/Models/Habit.swift

```diff
@@ -11,11 +11,40 @@ import Foundation
 
 // Habits are stored by category as a dictionary mapping habit titles to an array of
 // dates on which that habit was checked in [dates are strings of form "MMDDYYYY"].
-// The active habits are stored in an array of form: [ cardio habit, strength habit, mindfulness habit]
+// The active habits are stored as strings in an array of form: [ cardio habit, strength habit, mindfulness habit]
 struct Habit {
     let title: String
-    let dates: [Date]
+    var dates: [Date]
+    let type: HabitTrackingType
     
+    /// Number of consective days going back from today that this habit has been logged;
+    /// includes today if the habit has been logged today
+    var streak: Int {
+        if dates.count == 0 {
+            return 0
+        }
+        
+        var sortedDates = dates.sorted()
+        sortedDates.reverse()
+        
+        if !( sortedDates[0].isToday() || sortedDates[0].isYesterday() ) {
+            return 0
+        }
+        
+        var streak = 1
+
+        for i in 0..<(sortedDates.count-1) {
```

Comment:
prefer to do
`sortedDates.forEach`

Reply:
Wouldn't that be a bit tough here since we need to compare different indexes and we can't `break`? Any way I've tried to implement it this way comes out pretty hacky 😕